### PR TITLE
Added support for API Token to access finance_us_sec_public_company table data Closes #25

### LINKED ISSUE #25: Add support for API Token to access finance_us_sec_public_company table data
**Is your feature request related to a problem? Please describe.**
The `finance_us_sec_public_company` table uses the endpoint `https://api.iextrading.com/1.0/ref-data/symbols?format=csv` which now requires an API token for access.

**Describe the solution you'd like**
Add support for an `api_token` to the plugin configuration to be able to seamlessly access the data provided by the `finance_us_sec_public_company` table.

**Describe alternatives you've considered**
The endpoint `https://api.iextrading.com/1.0/ref-data/symbols?format=csv` now returns a `403 Forbidden error`.

**Additional context**
https://iexcloud.io/docs/api-basics/api-tokens

### PR DESCRIPTION
**Pull request by ParthaI**
# Example query results
<details>
  <summary>Results</summary>
  
```
> select * from finance_us_sec_public_company
+------------------------------------------------------------------------------------------------------------------+--------+------------+----------+------------+---------------------------+----------+--------------------------------+------------------+---------------->
| name                                                                                                             | symbol | cik        | currency | is_enabled | date                      | exchange | exchange_name                  | exchange_segment | exchange_segmen>
+------------------------------------------------------------------------------------------------------------------+--------+------------+----------+------------+---------------------------+----------+--------------------------------+------------------+---------------->
| Alcoa Corp                                                                                                       | AA     | 0001675149 | USD      | true       | 2023-11-17T05:30:00+05:30 | XNYS     | New York Stock Exchange Inc    | XNYS             | New York Stock >
| ATA Creativity Global - ADR                                                                                      | AACG   | 0001420529 | USD      | true       | 2023-11-17T05:30:00+05:30 | XNAS     | Nasdaq All Markets             | XNMS             | Nasdaq Nms Glob>
| AllianceBernstein Holding Lp - Unit                                                                              | AB     | 0000825313 | USD      | true       | 2023-11-17T05:30:00+05:30 | XNYS     | New York Stock Exchange Inc    | XNYS             | New York Stock >
| ACM Research Inc - Class A                                                                                       | ACMR   | 0001680062 | USD      | true       | 2023-11-17T05:30:00+05:30 | XNAS     | Nasdaq All Markets             | XNMS             | Nasdaq Nms Glob>
| Armada Acquisition Corp I - Units (1 Ord & 1/2 War)                                                              | AACIU  | 0001844817 | USD      | true       | 2023-11-17T05:30:00+05:30 | XNAS     | Nasdaq All Markets             | XNMS             | Nasdaq Nms Glob>
| Atlantic American Corp.                                                                                          | AAME   | 0000008177 | USD      | true       | 2023-11-17T05:30:00+05:30 | XNAS     | Nasdaq All Markets             | XNMS             | Nasdaq Nms Glob>
| Accel Entertainment Inc - Class A1                                                                               | ACEL   | 0001698991 | USD      | true       | 2023-11-17T05:30:00+05:30 | XNYS     | New York Stock Exchange Inc    | XNYS             | New York Stock >
| Aarons Company Inc (The)                                                                                         | AAN    | 0001821393 | USD      | true       | 2023-11-17T05:30:00+05:30 | XNYS     | New York Stock Exchange Inc    | XNYS             | New York Stock >
| Ares Acquisition Corporation II - Class A                                                                        | AACT   | 0001853138 | USD      | true       | 2023-11-17T05:30:00+05:30 | XNYS     | New York Stock Exchange Inc    | XNYS             | New York Stock >
| Arbor Realty Trust Inc. - 6.25% PRF PERPETUAL USD 25 - Ser E                                                     | ABR-E  | 0001253986 | USD      | true       | 2023-11-17T05:30:00+05:30
```
</details>

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -63,9 +63,13 @@ Installing the latest finance plugin will create a config file (`~/.steampipe/co
 ```hcl
 connection "finance" {
   plugin = "finance"
+
+  iex_api_token = "pk_3746625324373hrf7fa44d2983d34e"
 }
 ```
 
+- `iex_api_token`(Optional) - An additional header name and its corresponding value to be sent to the server. If the header name is configured, the value must not be empty.
```
Comment (cbruno10) [inline docs/index.md:71]:
```suggestion
- `iex_api_token` (Optional) - An additional header name and its corresponding value to be sent to the server. If the header name is configured, the value must not be empty.
```

Also wrong description?

File: config/finance.spc [hunk 2]
```diff
@@ -1,3 +1,8 @@
 connection "finance" {
   plugin = "finance"
+
+  # IEX Cloud authenticates your API requests using your account’s API tokens.
+  # https://iexcloud.io/docs/api-basics/api-tokens, https://iexcloud.io/documentation/reference/glossary.html#token-api-token
```
Comment (cbruno10) [inline config/finance.spc:5]:
```suggestion
  # https://iexcloud.io/documentation/reference/glossary.html#token-api-token
```

File: config/finance.spc [hunk 3]
```diff
@@ -1,3 +1,8 @@
 connection "finance" {
   plugin = "finance"
+
+  # IEX Cloud authenticates your API requests using your account’s API tokens.
```
Comment (cbruno10) [inline config/finance.spc:4]: We should mention this token is only used for the `finance_us_sec_public_company` table and that a publishable token is OK

File: config/finance.spc [hunk 4]
```diff
@@ -1,3 +1,8 @@
...
+  # https://iexcloud.io/docs/api-basics/api-tokens, https://iexcloud.io/documentation/reference/glossary.html#token-api-token
+  # Tokens can have data endpoint read access and read permissions for resources.
```
Comment (cbruno10) [inline config/finance.spc:6]:
```suggestion
```

File: finance/table_finance_us_sec_public_company.go [hunk 5]
```diff
@@ -19,18 +21,87 @@ func tableFinanceUsSecPublicCompany(ctx context.Context) *plugin.Table {
...
+	Region              string `json:"region"`
+	Symbol              string `json:"symbol"`
+	Type                string `json:"type"`
+	// Define other fields based on the API response
+}
+
 func listUsSecPublicCompany(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
-	companies, err := core.GetPublicCompaniesWithHeaders(map[string]string{"User-Agent": "Steampipe/v0.x"})
+
+	client := resty.New()
```
Comment (cbruno10) [inline finance/table_finance_us_sec_public_company.go:64]: Please try using a popular SDK if one exists

File: finance/table_finance_us_sec_public_company.go [hunk 6]
```diff
@@ -19,18 +21,87 @@ func tableFinanceUsSecPublicCompany(ctx context.Context) *plugin.Table {
...
+		}).
+		SetQueryParams(map[string]string{
+			"token": *config.IEXAPIToken,
+		}).
+		SetResult(&pageCompanies).
+		Get(apiUrl)
+
 	if err != nil {
-		plugin.Logger(ctx).Error("finance_us_sec_public_company.listUsSecPublicCompany", "query_error", err)
+		plugin.Logger(ctx).Error("finance_us_sec_public_company.listUsSecPublicCompany", "Error on request: ", err)
```
Comment (cbruno10) [inline finance/table_finance_us_sec_public_company.go:94]:
```suggestion
		plugin.Logger(ctx).Error("finance_us_sec_public_company.listUsSecPublicCompany", "query_error", err)
```

The second arg should be a snake cased error type

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@ParthaI Please see comments, thanks!

File: finance/table_finance_us_sec_public_company.go (hunk 5 above)
Comment (ParthaI) [reply finance/table_finance_us_sec_public_company.go]: Updated the package to make the API call.

**Comment by mingfang** [comment]
please merge

**Comment by kinghuang** [comment]
What's missing to merge this?

**Review by misraved — APPROVED** [review]

**Pull request merged**